framework.go

```diff
@@ -240,12 +252,22 @@ func (f *Framework) ApplyBuiltCluster(ctx context.Context, builtCluster *applica
 //
 //	wcClient, err := framework.WaitForClusterReady(timeoutCtx, "test-cluster", "default")
 func (f *Framework) WaitForClusterReady(ctx context.Context, clusterName string, namespace string) (*client.Client, error) {
-	err := wait.For(wait.IsClusterReadyCondition(ctx, f.MC(), clusterName, namespace), wait.WithContext(ctx), wait.WithInterval(10*time.Second))
+	c := f.getKubeConfigClient()
+
+	err := wait.For(wait.IsClusterReadyCondition(ctx, c, clusterName, namespace), wait.WithContext(ctx), wait.WithInterval(10*time.Second))
 	if err != nil {
 		return nil, err
 	}
 
-	return client.NewFromSecret(ctx, f.MC(), clusterName, namespace)
+	return client.NewFromClient(ctx, c, clusterName, namespace)
+}
+
+func (f *Framework) getKubeConfigClient() any {
```

Comment:
Please can we not use `any` type. If needed we sure use an interface.

Reply:
Don't worry, this is stil WIP :) I will be adding an interface for the kubeconfig client

Reply:
Ah! Sorry. I saw the request for review and dove in 😉

Reply:
Wanna ping me when it's good to go and I'll take a look?

Reply:
Yeah definitely! Sorry for the premature ping, I'm used to the draft button being the default :sweat_smile:

Reply:
No worries :) I like to keep on top of notifications when they come in. I didn't spot the WIP in the title.